ddtrace/internal/ci_visibility/recorder.py

```diff
@@ -206,6 +217,73 @@ def _agent_evp_proxy_is_available(self):
                 return True
         return False
 
+    @classmethod
+    def test_skipping_enabled(cls):
+        if not cls.enabled:
+            return False
+        return getattr(cls._instance, "_test_skipping_enabled_by_api", False)
+
+    @classmethod
+    def should_skip(cls, test, suite, module):
+        if not cls.enabled:
+            return False
+
+        return test in cls._instance._get_tests_to_skip(suite, module)
+
+    def _fetch_tests_to_skip(self):
+        # Make sure git uploading has finished
+        # this will block the thread until that happens
+        if self._git_client is not None:
+            self._git_client.shutdown()
+            self._git_client = None
+
+        payload = {
+            "data": {
+                "type": "test_params",
+                "attributes": {
+                    "service": self._service,
+                    "env": ddconfig.env,
+                    "repository_url": self._tags.get(ci.git.REPOSITORY_URL),
+                    "sha": self._tags.get(ci.git.COMMIT_SHA),
+                    "configurations": {},
+                },
+            }
+        }
+        payload["data"]["attributes"]["configurations"] = ci._get_runtime_and_os_metadata()
+
+        _headers = {
+            "dd-api-key": self._api_key,
+            "dd-application-key": self._app_key,
+            "Content-Type": "application/json",
+        }
+        if self._requests_mode == REQUESTS_MODE.EVP_PROXY_EVENTS:
+            url = EVP_PROXY_AGENT_BASE_PATH + SKIPPABLE_ENDPOINT
+            _headers = {EVP_SUBDOMAIN_HEADER_NAME: EVP_SUBDOMAIN_HEADER_API_VALUE}
+        elif self._requests_mode == REQUESTS_MODE.AGENTLESS_EVENTS:
+            url = "https://api." + self._dd_site + SKIPPABLE_ENDPOINT
+        else:
+            log.warning("Cannot make requests to skippable endpoint if mode is not agentless or evp proxy")
+            return
+
+        response = _do_request("POST", url, json.dumps(payload), _headers)
+
+        if response.status >= 400:
+            log.warning("Test skips request responded with status %d", response.status)
+            return
+        try:
+            parsed = json.loads(response.body)
+        except json.JSONDecodeError:
+            log.warning("Test skips request responded with invalid JSON '%s'", response.body)
```

Comment:
Can the response body contain PII or sensitive data?

Reply:
No, this is just a list of test suites and test names